function/bool_parameter_validator.go

```diff
@@ -0,0 +1,33 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package function
+
+import (
+	"context"
+
+	"github.com/hashicorp/terraform-plugin-framework/types"
+)
+
+// BoolValidator is a function validator for types.Bool parameters.
+type BoolValidator interface {
```

Comment:
The file name, `bool_parameter_validator.go` makes the intent clear, but I'm wondering whether more explicit naming might be warranted given that the `<Type>Validator`, and request and response types are all in the `function` package. Perhaps `BoolParameterValidator`, `BoolParameterValidatorRequest`, `BoolParameterValidatorResponse`?